internal/healthcheck/go_health.go

```diff
@@ -0,0 +1,78 @@
+package healthcheck
+
+import "github.com/InVisionApp/go-health"
+
+type GoHealthService struct {
+	Client HealthClient
+	jobs   []*HealthJob
+}
+
+type HealthClient interface {
+	AddChecks(cfgs []*health.Config) error
+	Start() error
+	Stop() error
+	State() (map[string]health.State, bool, error)
+}
+
+func (s *GoHealthService) Start() error {
+
+	cfgs := []*health.Config{}
+	for _, job := range s.jobs {
+		cfgs = append(cfgs, &health.Config{
+			Name:     job.Name,
+			Checker:  job.Checker,
+			Interval: job.Interval,
+		})
+	}
+	err := s.Client.AddChecks(cfgs)
+	if err != nil {
+		return err
+	}
+
+	return s.Client.Start()
+}
+
+func (s *GoHealthService) Stop() error {
+	return s.Client.Stop()
+}
+
+func (s *GoHealthService) Check() (*CheckResult, error) {
```

Comment:
I find myself do not understand "why" this function exists as well as other methods. For new joiner like me, it is not clear what to achieve with these methods. Minutes are taken to understand the whole idea.

Do we need to add comments that explain these things like some open source projects?
- Why (problem),
- what to achieve.